feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio.c

```diff
@@ -669,6 +674,150 @@ static void callback_Hotspot20_Icon_Config(ovsdb_update_monitor_t *mon,
 
 }
 
+enum {
+	WIF_APC_ENABLE,
+	__WIF_APC_MAX,
+};
+
+static const struct blobmsg_policy apc_enpolicy[__WIF_APC_MAX] = {
+		[WIF_APC_ENABLE] = { .name = "enabled", BLOBMSG_TYPE_BOOL },
+};
+
+const struct uci_blob_param_list apc_param = {
+	.n_params = __WIF_APC_MAX,
+	.params = apc_enpolicy,
+};
+
+void APC_config_update(struct schema_APC_Config *conf)
+{
+	struct uci_package *apc;
+	struct uci_context *uci;
+	struct blob_buf apcb = { };
+
+	LOGN("APC: APC_config_update");
+	uci = uci_alloc_context();
+
+	uci_load(uci, "apc", &apc);
+	blob_buf_init(&apcb, 0);
+
+	if (conf->enabled == true) {
+		blobmsg_add_bool(&apcb, "enabled", 1);
+		system("/etc/init.d/apc start");
+	}
+	else {
+		blobmsg_add_bool(&apcb, "enabled", 0);
+		system("/etc/init.d/apc stop");
+	}
+
+        blob_to_uci_section(uci, "apc", "apc", "apc",
+                            apcb.head, &apc_param, NULL);
+
+	uci_commit(uci, &apc, false);
+	uci_unload(uci, apc);
+}
+
+static void callback_APC_Config(ovsdb_update_monitor_t *mon,
+                                struct schema_APC_Config *old,
+                                struct schema_APC_Config *conf)
+{
+	if (mon->mon_type != OVSDB_UPDATE_DEL)
```

Comment:
What happens if the table is deleted while the service is enabled?  It will remain running.

A later reconfiguration will start the service a second time.  Will two copies run?  Will that cause a problem?

Reply:
If the table gets deleted the apc service is still running. apc service will not start a second time.

Reply:
If the table gets deleted, should the service still be running?